lib/CL/devices/devices.c

```diff
@@ -114,6 +142,43 @@ static unsigned devices_active = 0;
 
 static pocl_lock_t pocl_init_lock = POCL_LOCK_INITIALIZER;
 
+static void * pocl_device_handles[POCL_NUM_DEVICE_TYPES];
+
+static void
+getPoclDeviceLibPath (char *result, char *device_name)
+{
+  Dl_info info;
+  if (dladdr ((void *)getPoclDeviceLibPath, &info))
+    {
+      char const *soname = info.dli_fname;
+      strcpy (result, soname);
+      char *last_slash = strrchr (result, '/');
+      *(++last_slash) = '\0';
+      if (strlen (result) > 0)
+        {
+#ifdef ENABLE_POCL_BUILDING
+          if (pocl_get_bool_option ("POCL_BUILDING", 0))
+            {
+              strcat (result, "devices/");
+              strcat (result, device_name);
+              strcat (result, "/");
+#else
+          if (0)
```

Comment:
This is strange. Can you just make a macro derived from ENABLE_POCL_BUILDING a boolean 0/1 and guard using that directly?

Reply:
I've changed it a bit to match the file `lib/CL/pocl_llvm_build.cc`